build.gradle

```diff
@@ -190,15 +201,62 @@ subprojects {
 			}
 		}
 	}
+
 	signing {
-		sign publishing.publications.mavenJava
+		if(OSSRH_TARGET == 'remote') {
+			sign publishing.publications.mavenJava
+		}
 	}
-	publishMavenJavaPublicationToMavenRepository.doFirst {
-		if(OSSRH_URL == null) {
-			throw new GradleException("Please specify the variable 'ossrhUrl' in '~/.gradle/gradle.properties'.")
+
+	publish.doFirst {
+		if(OSSRH_TARGET != 'remote' && OSSRH_TARGET != 'local') {
+			throw new GradleException("Please specify the publication target with '-PossrhTarget=[remote|local]'.")
 		}
 		if(version == "unspecified") {
-			throw new GradleException("Please specify the version to release with '-Pversion=X.Y.Z'.")
+			throw new GradleException("Please specify the version to publish with '-Pversion=X.Y.Z'.")
+		}
+	}
+
+	// This task is created dynamically, upon a valid 'signing' definition.
+	if(tasks.findByName('signMavenJavaPublication')) {
+		signMavenJavaPublication.doFirst {
+			if(OSSRH_TARGET == 'remote') {
+				if(SIGNING_KEY_ID == null) {
+					throw new GradleException("Please specify the PGP signing key ID with 'signing.keyId' in '~/.gradle/gradle.properties'.")
+				}
+				// TODO Fix inconsistent naming:
+				// (1) In a PGP context, the term of "passphrase" is preferred over "password"
```

Comment:
I for one don't care if it is password or passphrase, it's both a text to unlock something. I vote for password, because it's what most people know and use.

Reply:
I believe the use of the term passphrase tries to emphasize that the password may contain whitespace, and to hint the user to use a good password/passphrase. Sounds reasonable but IMO it's the wrong approach.